ui/src/components/ClubsCarousel/index.tsx

```diff
@@ -1,37 +1,46 @@
 import styles from './styles.module.scss';
 import Carousel from 'react-multi-carousel';
 import 'react-multi-carousel/lib/styles.css';
-import {dummyClubLogo} from '../../assets';
-
-const data = [
-  {
-    name: 'TPC',
-    category: 'Technical',
-    logoUrl: dummyClubLogo,
-  },
-  {
-    name: 'Jazbaat',
-    category: 'Cultural',
-    logoUrl: dummyClubLogo,
-  },
-  {
-    name: 'Basketball',
-    category: 'Sports',
-    logoUrl: dummyClubLogo,
-  },
-  {
-    name: 'Saaz',
-    category: 'Cultural',
-    logoUrl: dummyClubLogo,
-  },
-  {
-    name: 'Badminton',
-    category: 'Sports',
-    logoUrl: dummyClubLogo,
-  },
-];
+import {useEffect, useState} from 'react';
+import {useNavigate} from 'react-router-dom';
 
 export const ClubsCarousel: React.FC = () => {
+  interface ClubsCarouselProps {
+    name: string;
+    logo: string;
+    category: string;
+  }
+  const [data, setData] = useState<ClubsCarouselProps[]>([]);
+  const token = window.sessionStorage.getItem('token');
+  const CLUBS_API_URL = 'http://127.0.0.1:8000/clubs/';
```

Comment:
Should not be hardcoded. Look at https://github.com/BitByte-TPC/gymkhana/blob/294716bc464488827e6d9e9d9aa55cb32ee02152/ui/src/api/googleOauth.ts#L34 for reference.